packages/nextjs/services/store/store.ts

```diff
@@ -1,13 +1,52 @@
-import { Abi, Address } from "viem";
+import { wagmiConnectors } from "../web3/wagmiConnectors";
+import { Abi, Address, Chain, createClient, http } from "viem";
+import { hardhat, mainnet } from "viem/chains";
+import { createConfig } from "wagmi";
 import create from "zustand";
 import scaffoldConfig from "~~/scaffold.config";
+import { getAlchemyHttpUrl, getTargetNetworks } from "~~/utils/scaffold-eth";
 import { ChainWithAttributes } from "~~/utils/scaffold-eth";
 
+const targetNetworks = getTargetNetworks();
+
+// We always want to have mainnet enabled (ENS resolution, ETH price, etc). But only once.
+export const enabledChains = targetNetworks.find((network: Chain) => network.id === 1)
+  ? targetNetworks
+  : ([...targetNetworks, mainnet] as const);
+
+const createWagmiClient = ({ chain }: { chain: Chain }) =>
+  createClient({
+    chain: {
+      id: chain.id,
+      name: chain.name,
+      nativeCurrency: chain.nativeCurrency,
+      rpcUrls: chain.rpcUrls,
+    },
+    transport: http(getAlchemyHttpUrl(chain.id)),
+    ...(chain.id !== (hardhat as Chain).id
+      ? {
+          pollingInterval: scaffoldConfig.pollingInterval,
+        }
+      : {}),
+  });
+
+export const wagmiConfig = createConfig({
+  chains: enabledChains as [Chain, ...Chain[]],
+  connectors: wagmiConnectors,
+  ssr: true,
+  client: createWagmiClient,
+});
```

Comment:
This stuff we already have at https://github.com/BuidlGuidl/abi.ninja/blob/3b075f30ce6b51d7e34887f80cb49eadc9c0aa73/packages/nextjs/services/web3/wagmiConfig.tsx#L15

Lets call it as `baseWagmiConfig` also maybe rename that file to `baseWagmiConfig` and import it from their

Reply:
Moved the declaration out of store.ts to baseWagmiConfig, also improved [typing at:](https://github.com/BuidlGuidl/abi.ninja/pull/129/commits/432187252f0d08aea19e6faa725d138c844234c7).